src/main/java/org/folio/qm/service/impl/MarcSpecificationServiceImpl.java

```diff
@@ -29,4 +34,19 @@ public SpecificationDto getSpecification(MarcFormat marcFormat) {
     }
     throw new IllegalArgumentException("Unknown format: " + marcFormat.name());
   }
+
+  @Override
+  public void updateSpecificationCache(SpecificationUpdatedEvent specificationUpdate) {
+    var specification = specificationStorageClient.getSpecification(specificationUpdate.specificationId());
+    var cache = cacheManager.getCache(SPECIFICATION_STORAGE_CACHE);
+
+    if (cache == null) {
+      log.warn("updateSpecificationCache:: no cache found for " + SPECIFICATION_STORAGE_CACHE);
+      return;
+    }
+
+    cache.put(specificationUpdate.tenantId() + ":" + specification.getProfile().name(), specification);
```

Comment:
We have cache by MarcFormat and not by FamilyProfile

Reply:
Yes, but they have same names